site/docs/build-event-protocol.md

```diff
@@ -0,0 +1,152 @@
+
+## The Build Event Protocol from 10,000 feet
+As mentioned before, the [Build Event Protocol] represents information about a build as events. A build event is a protocol buffer message consisting of a build event identifier, a set of child events and a payload.
+ 
+* __Build Event Identifier:__
+    Depending on the kind of build event, it might be an [opaque string] or [structured information] revealing more about the build event. A build event identifier is unique within a build.
+
+* __Children:__ A build event may announce other build events, by including their build event identifiers in its [children field]. For example, the `PatternExpanded` build event announces the targets it expands to as children. The main purpose of child events is to ensure that no event except for the initial event is posted unexpectedly.
+
+* __Payload:__ The payload contains structured information about a build event, encoded as a protocol buffer message specific to that event.
+
+### Build Event Graph
+All build events form a [directed acyclic graph] through their parent and child relationship. Every build event except for the initial build event has one or more parent events. Please note that not all parent events of a child event must necessarily be posted before it. When a build is complete (succeeded or failed) all announced events will have been posted. In case of errors i.e. Bazel crashed or a failed network transport, some announced build events may never be posted.
+
+## The Build Event Protocol by Example
+The full specification of the [Build Event Protocol] can be found in its protocol buffer definition and describing it here is beyond the scope of this document. However, it might be helpful to build up some intuition before looking at the specification.
+ 
+Consider a simple Bazel workspace that consists of two empty shell scripts `foo.sh` and `foo_test.sh` and a `BUILD` file:
+
+```bash
+sh_binary(
+    name = "foo",
+    srcs = ["foo.sh"],
+)
+ 
+sh_library(
+    name = "foo_lib",
+    data = [":foo"],
+)
+ 
+sh_test(
+    name = "foo_test",
+    srcs = ["foo_test.sh"],
+    deps = [":foo_lib"],
+)
+```
+
+ 
+When running `bazel test …` on this project the build graph of the generated build events will resemble the graph below. The arrows indicate the aforementioned parent and child relationship. Please note that some build events and most fields have been omitted for briefness.
+
+![bep-graph]
+
+Initially, a `BuildStarted` event is published. The event informs us that the build was invoked through the `bazel test` command and it also announces five child events: `OptionsParsed`, `WorkspaceStatus`, `CommandLine`, `PatternExpanded` and `Progress`. The first three events provide information about how Bazel was invoked. The `PatternExpanded` build event provides insight into which specific targets the `…` pattern expanded to: `//:foo`, `//:foo_lib` and `//:foo_test`. It does so by declaring three `TargetConfigured` events as children.
```

Comment:
The pattersn is actually `...`, i.e., three literal dots, not an ellipsis symbol.